### DISCUSSION TITLE
[IAC] Configure Postal as-code

### DISCUSSION THREAD

**Discussion by arthurzenika** [postalserver/postal#2597] [Feature suggestions]
We are looking into automation of the deployment of postal, and were wondering if there were any other users that were using automation for the installation and configuration of postal. 

We have bits of Ansible for the installation part, but we don't want to re-invent the wheel if somethings exists. 

Also, for the configuration inside postal (organisations, servers, ip pools, etc ) we'd love to be able to automate that (terraform, ansible, etc), is there existing initiatives ? Right now we are looking at APIs and maybe even some SQL automation or partial backup/restore to avoid manual configuration in the web UI.

**Comment by willpower232** [comment]
Unfortunately Postal is intended to be used with a web UI for now so almost all of the configuration is in MySQL, however if you are able to seed in the correct databases from an empty state then you could probably automate the whole process.
